src/Core/Settings.h

```diff
@@ -700,6 +700,7 @@ class IColumn;
     M(Bool, force_aggregate_partitions_independently, false, "Force the use of optimization when it is applicable, but heuristics decided not to use it", 0) \
     M(UInt64, max_number_of_partitions_for_independent_aggregation, 128, "Maximal number of partitions in table to apply optimization", 0) \
     M(Float, min_hit_rate_to_use_consecutive_keys_optimization, 0.5, "Minimal hit rate of a cache which is used for consecutive keys optimization in aggregation to keep it enabled", 0) \
+    M(Bool, enable_fast_materialize_ttl, false, "Enable fast materialize ttl optimization.", 0) \
```

Comment:
Why hide this change behind a standalone setting? It doesn't look like it's breaking some old behavior, does it?

Reply:
Because of the addition of `ALTER TABLE tab MATERIALIZE TTL [ttl_delta]` syntax, if the TTL is modified during the rolling upgrade process, the old version will not be able to parse and fail.
If there are better suggestions, I'm willing to implement them.